packages/hash/external-services/docker-compose.prod.yml

```diff
@@ -0,0 +1,206 @@
+      POSTGRES_PASSWORD: "${POSTGRES_PASSWORD}"
+      HASH_KRATOS_PG_USER: "${HASH_KRATOS_PG_USER}"
+      HASH_KRATOS_PG_PASSWORD: "${HASH_KRATOS_PG_PASSWORD}"
+      HASH_KRATOS_PG_DATABASE: "${HASH_KRATOS_PG_DEV_DATABASE}"
+      HASH_GRAPH_PG_USER: "${HASH_GRAPH_PG_USER}"
+      HASH_GRAPH_PG_PASSWORD: "${HASH_GRAPH_PG_PASSWORD}"
+      HASH_GRAPH_PG_DATABASE: "${HASH_GRAPH_PG_DEV_DATABASE}"
+      HASH_GRAPH_REALTIME_PG_USER: "${HASH_GRAPH_REALTIME_PG_USER}"
+      HASH_GRAPH_REALTIME_PG_PASSWORD: "${HASH_GRAPH_REALTIME_PG_PASSWORD}"
+    # ports:
+    #   - "5432:5432"
+    volumes:
+      - hash-postgres-data:/var/lib/postgresql/data
+      - ./postgres/postgresql.conf:/etc/postgresql/postgresql.conf:ro
+      - ./postgres/init-user-db.sh:/docker-entrypoint-initdb.d/init-user-db.sh:ro
+    healthcheck:
+      test: ["CMD-SHELL", "pg_isready --username ${POSTGRES_USER}"]
+      interval: 2s
+      timeout: 2s
+      retries: 5
+    command: -c 'config_file=/etc/postgresql/postgresql.conf'
+
+  hash-graph-migrate:
+    build:
+      context: ../../graph
+      dockerfile: deployment/migrations/Dockerfile
+    depends_on:
+      postgres:
+        condition: service_healthy
+    environment:
+      HASH_GRAPH_PG_MIGRATION_URL: "postgres://${POSTGRES_USER}:${POSTGRES_PASSWORD}@postgres:5432/${HASH_GRAPH_PG_DEV_DATABASE}"
+
+  hash-graph:
+    build:
+      dockerfile: deployment/graph/Dockerfile
+      context: ../../graph
+    init: true
+    depends_on:
+      postgres:
+        condition: service_healthy
+      hash-graph-migrate:
+        condition: service_completed_successfully
+    environment:
+      HASH_GRAPH_PG_USER: "${HASH_GRAPH_PG_USER}"
+      HASH_GRAPH_PG_PASSWORD: "${HASH_GRAPH_PG_PASSWORD}"
+      HASH_GRAPH_PG_HOST: "postgres"
+      HASH_GRAPH_PG_PORT: "5432"
+      HASH_GRAPH_PG_DATABASE: "${HASH_GRAPH_PG_DEV_DATABASE}"
+      HASH_GRAPH_LOG_FORMAT: "${HASH_GRAPH_LOG_FORMAT:-pretty}"
+      HASH_GRAPH_ALLOWED_URL_DOMAIN_PATTERN: "${FRONTEND_URL}/@(?P<shortname>[\\w-]+)/types/(?P<kind>(?:data-type)|(?:property-type)|(?:entity-type)|(?:link-type))/[\\w-]+/"
+      RUST_LOG: "${HASH_GRAPH_LOG_LEVEL:-graph=trace,hash_graph=trace,tokio_postgres=debug}"
+      RUST_BACKTRACE: 1
+    ports:
+      - "4000:4000"
+
+  kratos-migrate:
+    image: oryd/kratos:v0.10.1
+    depends_on:
+      postgres:
+        condition: service_healthy
+    read_only: true
+    volumes:
+      - type: bind
+        source: ./kratos
+        target: /etc/config/kratos
+    environment:
+      - DSN=postgres://${POSTGRES_USER}:${POSTGRES_PASSWORD}@postgres:5432/${HASH_KRATOS_PG_DEV_DATABASE}
+    command: -c /etc/config/kratos/kratos.dev.yml migrate sql -e --yes
+    security_opt:
+      - no-new-privileges:true
+
+  kratos:
+    image: oryd/kratos:v0.10.1
+    depends_on:
+      - kratos-migrate
+    read_only: true
+    # ports:
+    #   - "4433:4433" # public, doesn't need to be exposed
+    #   - "4434:4434" # admin, we don't want to expose this
```

Comment:
I'm not sure why this is commented out, can you explain why the ports aren't exposed?

Reply:
Yes, thanks. I've added comments in 7b4d8c5907f1eba134673d9e064efdbd092d5405